### DISCUSSION TITLE
How can I run ADB commands using Shizuku?

### DISCUSSION THREAD

**Discussion by hussam-alhassan** [RikkaApps/Shizuku#404] [Q&A]
I want my app to be able to execute a specific ADB command, and I read before that you can do it through Shizuku but I can't find anything on that.

The command I'm trying to execute is "adb shell pm install-existing --user 95 com.package.name"

Is this possible at all?

**Comment by SuperCowProducts** [comment]
I used to do this (running the commands separately):
```bash
sh rish
pm install-existing --user 95 com.package.name
```

In theory, `sh rish` should start a new interactive `adb` subshell (which means you mustn't prepend  `adb shell`).

I actually came here as I was having issues despite having done so successfully in the past; another alternative would be to connect it over wireless debugging.

Feel free to ask more questions, I only know a tiny bit about all of this but I may be able to help!
